.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+on:
+  push:
+    branches: [ staging, trying, master ]
+  pull_request:
+
+name: Continuous integration
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+          - thumbv6m-none-eabi
+          - thumbv7m-none-eabi
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          target: ${{ matrix.target }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --examples
```

Comment:
```suggestion
          args: --examples --target ${{ matrix.target }}
```

Reply:
Thanks, do you happen to know why clippy isn't running ?

Reply:
It uses `pull_request_target`, which means that it'll use the GHA configuration from master, not this PR

Reply:
With clippy set to `pull_request_target`, the workflow uses the version of the file that exists in the base branch, but it doesn't exist yet in master so nothing runs. Once the PR is merged it "should" work.

Since this PR is not from a fork, you could change `pull_request_target` to just `pull_request` for a commit, check clippy runs and the YAML works, then change it back to `pull_request_target` before merging.

Reply:
Should I add `pull_request` together with `pull_request_target` ?

Reply:
No, after this is merged that would just result in it running twice on PRs.

Reply:
> No, after this is merged that would just result in it running twice on PRs.

Ops, sorry, when I commented the UI wasn't updated with your comments